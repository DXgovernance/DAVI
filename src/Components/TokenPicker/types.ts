import { TokenWithBalance } from 'hooks/Guilds/ether-swr/erc20/useAllERC20Balances';

export type TokenWithBalanceIndexable = TokenWithBalance & { id: string };

export interface TokenPickerProps {
  walletAddress?: string;
  isOpen: boolean;
  onClose: () => void;
  onSelect: (tokenAddress: string) => void;
}
