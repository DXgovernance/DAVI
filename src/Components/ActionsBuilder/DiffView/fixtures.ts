export const mockCodeOriginal = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;

export const mockCodeUpdated = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;
