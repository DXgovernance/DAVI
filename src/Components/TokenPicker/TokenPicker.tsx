import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import {
  TokenWithBalance,
  useAllERC20Balances,
} from 'hooks/Guilds/ether-swr/erc20/useAllERC20Balances';
import { TokenPickerProps } from './types';
import { Picker } from 'Components/Primitives/Forms/Picker';
import { ethers } from 'ethers';
import { resolveUri } from 'utils/url';
import Avatar from 'old-components/Guilds/Avatar';
import { useAccount } from 'wagmi';

const TokenPicker: React.FC<TokenPickerProps> = ({
  walletAddress,
  isOpen,
  onSelect,
  onClose,
}) => {
  const { t } = useTranslation();
  const { address } = useAccount();
  const { data } = useAllERC20Balances(walletAddress || address);

  const handleSelect = (option: TokenWithBalance) => {
    onSelect(option.address);
  };

  // "Translate" token data to fields needed in the Picker component
  const tokens = useMemo(() => {
    return data.map(token => {
      let formattedBalance = (
        token.balance !== undefined
          ? ethers.utils.formatUnits(token?.balance, token?.decimals)
          : '-'
      ).slice(0, -2);

      return {
        ...token,
        id: token?.address,
        title: token?.symbol,
        subtitle: token?.name,
        value: token?.address,
        rightData: formattedBalance,
        icon: (
          <Avatar
            src={resolveUri(token?.logoURI)}
            defaultSeed={token?.address}
            size={28}
          />
        ),
      };
    });
  }, [data]);

  return (
    <Picker
      data={tokens}
      header={t('selectAToken')}
      isOpen={isOpen}
      onSelect={handleSelect}
      onClose={onClose}
    />
  );
};

export default TokenPicker;
