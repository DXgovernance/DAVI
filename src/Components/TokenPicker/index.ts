export { default as TokenPicker } from './TokenPicker';
