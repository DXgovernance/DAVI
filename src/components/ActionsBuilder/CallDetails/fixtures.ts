import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
