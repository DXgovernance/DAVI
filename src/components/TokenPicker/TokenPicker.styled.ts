import styled from 'styled-components';

export const TokenPickerContainer = styled.div`
  margin: 2rem;
`;

export const SearchWrapper = styled.div`
  display: flex;
  margin-bottom: 1.5rem;
`;

export const TokenList = styled.div`
  display: flex;
  flex-direction: column;
`;
