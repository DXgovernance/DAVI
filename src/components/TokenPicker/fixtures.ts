export const data = [
  {
    address: '0xB58b358Be2B16Cb8992fc7340F92127083de33cE',
    name: 'SWAPR Guild',
    decimals: 18,
    symbol: 'SWPR',
    fetchPrice: true,
    logoURI: 'https://s2.coinmarketcap.com/static/img/coins/200x200/5589.png',
    chainId: 31337,
    balance: {
      type: 'BigNumber',
      hex: '0x8ac7230489e80000',
    },
  },
  {
    address: '0xfFb1cd0F95368DDd06D556161c5D3d9f0f4Fe6d2',
    name: 'DXdao on Localhost',
    decimals: 18,
    symbol: 'DXD',
    fetchPrice: true,
    logoURI: 'https://s2.coinmarketcap.com/static/img/coins/200x200/5589.png',
    chainId: 31337,
    balance: {
      type: 'BigNumber',
      hex: '0x00',
    },
  },
  {
    address: '0x7BC0DEDaFd60611D430c89353CC30e1a11b90AC7',
    name: 'REP Guild Token on Localhost',
    decimals: 18,
    symbol: 'RGT',
    fetchPrice: true,
    logoURI: 'https://s2.coinmarketcap.com/static/img/coins/200x200/5589.png',
    chainId: 31337,
    balance: {
      type: 'BigNumber',
      hex: '0x00',
    },
  },
];
