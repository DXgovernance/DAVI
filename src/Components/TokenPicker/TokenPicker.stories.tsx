import { ComponentStory, ComponentMeta } from '@storybook/react';
import { TokenPicker } from 'Components';
// import { data } from './fixtures';

export default {
  title: 'Components/TokenPicker',
  component: TokenPicker,
} as ComponentMeta<typeof TokenPicker>;

const Template: ComponentStory<typeof TokenPicker> = args => (
  <TokenPicker {...args} />
);

export const Default = Template.bind({});
Default.args = { isOpen: true, onClose: () => {}, onSelect: () => {} };
Default.storyName = 'TokenPicker';
