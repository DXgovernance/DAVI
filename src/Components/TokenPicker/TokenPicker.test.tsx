import TokenPicker from './TokenPicker';
import { render } from 'utils/tests';
import { useAllERC20Balances } from 'hooks/Guilds/ether-swr/erc20/useAllERC20Balances';
import { data } from './fixtures';

jest.mock('hooks/Guilds/ether-swr/erc20/useAllERC20Balances', () => ({
  useAllERC20Balances: jest.fn(),
}));

jest.mock('contexts/index', () => jest.fn());

const mockedERC20Balances = useAllERC20Balances as jest.Mock;

describe('TokenPicker', () => {
  let props;
  beforeEach(() => {
    props = {
      isOpen: true,
      onClose: jest.fn(),
      onSelect: () => jest.fn(),
    };
  });

  it('Should match snapshot', () => {
    console.error = jest.fn();

    mockedERC20Balances.mockImplementation(() => ({
      data,
    }));

    const { container } = render(<TokenPicker {...props} />, {
      container: document.body,
    });
    expect(container).toMatchSnapshot();
  });
});
