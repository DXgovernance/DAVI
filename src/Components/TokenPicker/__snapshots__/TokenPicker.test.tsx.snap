// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenPicker Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-jONnzC hUvaRT backdropModal-3"
    >
      <div
        class="sc-aemIz fpuFHE"
      >
        <div
          class="sc-hlTwrV kXDWEa"
        >
          
           
          <div
            class="sc-eLgNKc GtOyU"
          >
            <h2
              class="sc-fFucqa sc-kIeSZW fhzUrS NYJKg"
            >
              selectAToken
            </h2>
            <svg
              class="sc-hOqruk cOtmWQ"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="18"
                x2="6"
                y1="6"
                y2="18"
              />
              <line
                x1="6"
                x2="18"
                y1="6"
                y2="18"
              />
            </svg>
          </div>
          <div
            class="sc-dacGJm logYbD"
          >
            <div
              class="sc-dFJtaz bBfQcN"
            >
              <div
                class="sc-bsipwG dACkam"
              >
                <div
                  class="sc-gsTEea jwbCic"
                >
                  <div
                    class="sc-dlfnuX jRxgXa"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="11"
                        cy="11"
                        r="8"
                      />
                      <line
                        x1="21"
                        x2="16.65"
                        y1="21"
                        y2="16.65"
                      />
                    </svg>
                  </div>
                  <input
                    class="sc-bdfBQB wpJaf"
                    placeholder="searchToken"
                    value=""
                  />
                  <div
                    class="sc-dlfnuX gJuvwh"
                  />
                </div>
              </div>
              <div
                class="sc-gInthZ iSfbIF"
              >
                <div
                  class="sc-GTVdH eWmvTz"
                >
                  <div
                    class="sc-gsxoIA csYcEO"
                  >
                    <div
                      class="sc-bXDkGd ZAcVe"
                    >
                      <img
                        alt="Avatar"
                        class="sc-jSgvzq sc-fubCzh iUxzmp jqAwTz"
                        data-testid="avatar"
                        size="28"
                        src="https://s2.coinmarketcap.com/static/img/coins/200x200/5589.png"
                      />
                    </div>
                    <div>
                      <div
                        class="sc-eGCaLh gdvMjc"
                      >
                        SWPR
                      </div>
                      <div
                        class="sc-ctaXho bKuKZN"
                      >
                        SWAPR Guild
                      </div>
                    </div>
                  </div>
                  <div>
                    10
                  </div>
                </div>
                <div
                  class="sc-GTVdH eWmvTz"
                >
                  <div
                    class="sc-gsxoIA csYcEO"
                  >
                    <div
                      class="sc-bXDkGd ZAcVe"
                    >
                      <img
                        alt="Avatar"
                        class="sc-jSgvzq sc-fubCzh iUxzmp jqAwTz"
                        data-testid="avatar"
                        size="28"
                        src="https://s2.coinmarketcap.com/static/img/coins/200x200/5589.png"
                      />
                    </div>
                    <div>
                      <div
                        class="sc-eGCaLh gdvMjc"
                      >
                        DXD
                      </div>
                      <div
                        class="sc-ctaXho bKuKZN"
                      >
                        DXdao on Localhost
                      </div>
                    </div>
                  </div>
                  <div>
                    0
                  </div>
                </div>
                <div
                  class="sc-GTVdH eWmvTz"
                >
                  <div
                    class="sc-gsxoIA csYcEO"
                  >
                    <div
                      class="sc-bXDkGd ZAcVe"
                    >
                      <img
                        alt="Avatar"
                        class="sc-jSgvzq sc-fubCzh iUxzmp jqAwTz"
                        data-testid="avatar"
                        size="28"
                        src="https://s2.coinmarketcap.com/static/img/coins/200x200/5589.png"
                      />
                    </div>
                    <div>
                      <div
                        class="sc-eGCaLh gdvMjc"
                      >
                        RGT
                      </div>
                      <div
                        class="sc-ctaXho bKuKZN"
                      >
                        REP Guild Token on Localhost
                      </div>
                    </div>
                  </div>
                  <div>
                    0
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
   
</body>
`;
